models/heartbeat.go

```diff
@@ -40,15 +34,12 @@ func (h *Heartbeat) Valid() bool {
 }
 
 func (h *Heartbeat) Augment(languageMappings map[string]string) {
-	groups := languageRegex.FindAllStringSubmatch(h.Entity, -1)
-	if len(groups) == 0 || len(groups[0]) != 2 {
-		return
-	}
-	ending := groups[0][1]
-	if _, ok := languageMappings[ending]; !ok {
-		return
+	for ending, value := range languageMappings {
+		if strings.HasSuffix(h.Entity, "."+ending) {
```

Comment:
Brief side note here: you could have used `fmt.Sptinf()`, which looks a bit cleaner to me, but might actually be slower, not sure.